RESTWS-721: Fix modules producing ref errors due to swagger exclusion

tomtom5152 opened the pull request:
Fix for ticket [RESTWS-721](https://issues.openmrs.org/browse/RESTWS-721). 
Add failing test checking for appropriate method calls and definitions in the compiled swagger.
Implement a fix deleting the offending code to allow tests to pass.

Builds on-top of #342 as changes required for tests to work.

dkayiwa commented:
@tomtom5152 do you mind taking a second look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

tomtom5152 commented:
Which bit specifically, I thought I had got everything sorted?

dkayiwa commented:
Reference the ticket. The summary/title of the pull request should contain the ticket's JIRA id e.g 'TRUNK-4196: Fix some issue' and you should include the link to the ticket in the description field. If you are working on a sub task of another parent ticket or story, the ticket id to use in the summary is that of the sub task and not that of the parent. The pull request title should also be a good summary of the work done. If you are not sure of what to put here, just use the ticket title.

tomtom5152 commented:
My apologies, I'd copied one of the other PRs currently open which is where I had gone wrong I think, got mixed up with what was wanted.
I've fixed the description and title as requested, and also added a few fixes for some travis errors that I hadn't seen due to how I was running the tests (IDE running). Full `mvn clean install` now passes locally.

tomtom5152 commented:
Rebased onto master following merge conflicts arising.
